app/src/main/java/zapsolutions/zap/util/UserGuardian.java

```diff
@@ -166,6 +167,17 @@ public void securityTooMuchMoney() {
         showGuardianDialog(adb);
     }
 
+    /**
+     * Warn the user if he is trying to connect to a remote BtcPay server.
+     */
+    public void securityConnectToBtcPay(String host) {
+        mCurrentDialogName = BTC_PAY_CONNECT;
+        AlertDialog.Builder adb = createDontShowAgainDialog(true);
+        String message = mContext.getResources().getString(R.string.guardian_btcPayConnect, host);
+        adb.setMessage(message);
+        showGuardianDialog(adb);
+    }
+
```

Comment:
Why do we warn the user before connecting to BTCPay Server?
Of course he wants to connect, that's why he scanned/pasted the connection in the first place.

If we do it for security reason, is this really a thing? Would someone try to trick you to connect to his node?

Reply:
In the desktop app we require the user to verify the host that we are about to connect to when setting up a new connection. This is important incase the user is copy/pasting the connection string in order to protect against some kind kind of malware / MITM attack where the clipboard content is messed with and the user ends up trying to connect to a malicious server instead of the one they thought they were connecting to. Asking the user to confirm the hostname gives them a chance to verify before connecting

Reply:
Okay, confirming one short prompt is ok for UX, so we can keep it. 

The same applies for a normal lndconnect string (no BTCPay) I guess. So we will have to add this later, too.